modules/dxp/apps/saml/saml-opensaml-integration/src/main/java/com/liferay/saml/opensaml/integration/internal/servlet/profile/WebSsoProfileImpl.java

```diff
@@ -1935,12 +1981,20 @@ private void _redirectToLogin(
 			if ((samlMessageInfoContext != null) &&
 				(samlMessageInfoContext.getMessageId() != null)) {
 
+				_bindSamlSsoRequestContext(
+					samlMessageInfoContext.getMessageId(), httpSession,
+					samlSsoRequestContext);
+
 				redirectSB.append("?saml_message_id=");
 				redirectSB.append(
 					URLCodec.encodeURL(samlMessageInfoContext.getMessageId()));
 			}
 		}
 		else if (samlPeerEntityContext.getEntityId() != null) {
+			_bindSamlSsoRequestContext(
+				"idpInitiated", httpSession,
```

Comment:
I think this may be one the of SF issues; try moving `samlSsoRequestContext` to the same line as the other arguments.